main.cpp

```diff
@@ -50,6 +50,14 @@ void VacuumDB(const string& db) { RetrySystem("sqlite3 " + db + " 'VACUUM;'"); }
 
 #define BACKUP_DIR "/var/tmp/"
 void BackupDB(const string& dbPath) {
+
+    // Initialize sqlite3 pointer '_db'
+    sqlite3* _db = NULL;
+
+    // Open database connection to prevent checkpointing
+    SINFO("Opening connection to " << dbPath);
+    SASSERT(sqlite3_open_v2(dbPath.c_str(), &_db, SQLITE_OPEN_READONLY, NULL));
```

Comment:
We might want to pass `SQLITE_OPEN_NOMUTEX` here as well. I can't find a defined behavior anywhere for what happens if we have the connection still open in another thread and then open it here in singlethreaded mode. @tylerkaraszewski do you have any idea on this?